repair/repair_meta.hh

```diff
@@ -0,0 +1,415 @@
+/*
```

Comment:
Again, a huge amount of code moved from .cc to .hh, which is unfortunate.
I don't know whether the new "repair_meta" idea makes sense and whether that should be its name (I have no idea what "repair meta" is supposed to mean - is it for tracking on going repairs? Or something else? I always like to have a comment in files explaining what they are about). I hope @asias can comment on the logic of having those separate files which you created and their names.

Reply:
The split makes sense to me. repair meta is the metadata for repair instance on the nodes participating the repair.